block/state.go

```diff
@@ -0,0 +1,123 @@
+		AppHash:         state.AppHash,
+		LastValidators:  state.LastValidators.Copy(),
+		LastStoreHeight: state.LastStoreHeight,
+
+		LastResultsHash: state.LastResultsHash,
+		BaseHeight:      state.BaseHeight,
+	}
+
+	return s, nil
+}
+
+func (e *BlockExecutor) UpdateStateAfterInitChain(s *types.State, res *abci.ResponseInitChain, validators []*tmtypes.Validator) {
+	// If the app did not return an app hash, we keep the one set from the genesis doc in
+	// the state. We don't set appHash since we don't want the genesis doc app hash
+	// recorded in the genesis block. We should probably just remove GenesisDoc.AppHash.
+	if len(res.AppHash) > 0 {
+		copy(s.AppHash[:], res.AppHash)
+	}
+
+	//The validators after initChain must be greater than zero, otherwise this state is not loadable
+	if len(validators) <= 0 {
+		panic("Validators must be greater than zero")
+	}
+
+	if res.ConsensusParams != nil {
+		params := res.ConsensusParams
+		if params.Block != nil {
+			s.ConsensusParams.Block.MaxBytes = params.Block.MaxBytes
+			s.ConsensusParams.Block.MaxGas = params.Block.MaxGas
+		}
+		if params.Evidence != nil {
+			s.ConsensusParams.Evidence.MaxAgeNumBlocks = params.Evidence.MaxAgeNumBlocks
+			s.ConsensusParams.Evidence.MaxAgeDuration = params.Evidence.MaxAgeDuration
+			s.ConsensusParams.Evidence.MaxBytes = params.Evidence.MaxBytes
+		}
+		if params.Validator != nil {
+			// Copy params.Validator.PubkeyTypes, and set result's value to the copy.
+			// This avoids having to initialize the slice to 0 values, and then write to it again.
+			s.ConsensusParams.Validator.PubKeyTypes = append([]string{}, params.Validator.PubKeyTypes...)
+		}
+		if params.Version != nil {
+			s.ConsensusParams.Version.AppVersion = params.Version.AppVersion
+		}
+		s.Version.Consensus.App = s.ConsensusParams.Version.AppVersion
+	}
+	// We update the last results hash with the empty hash, to conform with RFC-6962.
+	copy(s.LastResultsHash[:], merkle.HashFromByteSlices(nil))
+
+	// Set the validators in the state
+	s.Validators = tmtypes.NewValidatorSet(validators).CopyIncrementProposerPriority(1)
+	s.NextValidators = s.Validators.Copy()
+	s.LastValidators = s.Validators.Copy()
+}
+
+// UpdateStateFromResponses updates state based on the ABCIResponses.
+func (e *BlockExecutor) UpdateStateFromResponses(resp *tmstate.ABCIResponses, state types.State, block *types.Block) (types.State, error) {
+	//Dymint ignores any setValidator responses from the app, as it is manages the validator set based on the settlement consensus
+	//TODO: this will be changed when supporting multiple sequencers from the hub
+	validatorUpdates := []*tmtypes.Validator{}
+
+	if state.ConsensusParams.Block.MaxBytes == 0 {
+		e.logger.Error("maxBytes=0", "state.ConsensusParams.Block", state.ConsensusParams.Block)
+	}
+
+	state, err := e.updateState(state, block, resp, validatorUpdates)
+	if err != nil {
+		return types.State{}, err
+	}
+
+	return state, nil
+}
```

Comment:
did these functions change? I'm guessing not

Reply:
good guess (: